Add Kubernetes support

etiennetremel opened the pull request:
This PR add Kubernetes support to ImageWolf.

Make use of DaemonSet and headless Service for peer discovery.

Commit:
Add Kubernetes support

hamhut1066 commented:
Hi @etiennetremel, I've made some changes to your pull request regarding stability and reproducibility of the build. Everything looks good though, and the documentation that you have included is clear and easy to follow. You can find my changes here:
[link to comparison](https://github.com/etiennetremel/ImageWolf/compare/add-kubernetes-support...ContainerSolutions:etiennetremel-add-kubernetes-support?expand=1)

which, as I can't commit directly to your PR I have put on a branch to be folded in by you.

Commit:
Apply Docker build cleanup from @hamhut1066

- Reformat the Kubernetes work to built a simple image.
- Work around issues with Docker versioning
- transition to using the _docker_ static binary

etiennetremel commented:
@hamhut1066 thanks a lot, I just applied your changes to the PR
